webapp/app/elster_client/elster_client.py

```diff
@@ -211,13 +216,22 @@ def _generate_est_request_data(form_data, year=2020):
 
     if adapted_form_data.get('steuernummer_exists') == 'no' and adapted_form_data.get('request_new_tax_number'):
         adapted_form_data['submission_without_tax_nr'] = True
+        
+    digitally_signed = bool(current_user.unlock_code_hashed is not None)
 
     if not current_user.is_active:
-        # no non-active user should come until here, but they should certainly not be able to send a tax
+        # no non-active user should come until here, but we want to log that as an error
+        logger.error('Elster_Client: Non-active user tries to send tax declaration.')
+
+    if not digitally_signed:
+        logger.warning('Elster_Client: User without unlock code tries to send tax declaration.')
+        # no user should come until that point without an unlock code, but they should certainly not be able to send a tax declaration
         logout_user()
+        raise TaxDeclarationNotDigitallySigned
+    
```

Comment:
@SannyNguyenHung: Do you think we should also logout the user and raise the error for the non active user or is it enough that we log the error?

Reply:
I think we should not log out the user, because we do not ask if he is authenticated.

Reply:
Believe it is enough just to log :)